File: frontend/src/types/order.ts
export interface Order {
  ot_idx: number;
  ot_code?: string;
  ot_ori_code?: string;
  mt_idx?: number;
  mt_id?: string;
  ot_title?: string;
  ot_pay_type?: string;
  ot_sdate?: string;
  ot_edate?: string;
  ot_status?: number;
  ot_sprice?: number;
  ot_use_coupon?: number;
  ot_coupon_idx?: number;
  ot_price?: number;
  ot_price_b?: number;
  ot_pay_bank?: string;
  ot_rsp_imp_uid?: string;
  ct_cancel_amount?: number;
  ct_cancel_reson?: string;
  ct_refund_holder?: string;
  ct_refund_bank?: string;
  ct_refund_account?: string;
  ot_wdate?: string;
  ot_pdate?: string;
  ot_cdate?: string;
  ot_ccdate?: string;
  ot_refund_info?: string;
  ot_show?: string;
  ot_memo?: string;
  ot_udate?: string;
  ot_cash_name?: string;
  ot_cash_hp?: string;
  ot_cash_chk?: string;
  vbank_date?: string;
}

export interface OrderListResponse {
  orders: Order[];
  total: number;
  page: number;
  size: number;
  total_pages: number;
}

export interface OrderSummary {
  total_orders: number;
  total_amount: number;
  completed_orders: number;
  pending_orders: number;
  cancelled_orders: number;
  this_month_amount: number;
  average_amount: number;
}

export interface OrderFilter {
  status?: number;
  pay_type?: string;
  start_date?: string;
  end_date?: string;
  page?: number;
  size?: number;
}

// 결제 방법 타입
export enum PaymentType {
  NEW_SIGNUP = '1',  // 신규가입
  APP_PAYMENT = '2', // 앱결제
  COUPON = '3',      // 쿠폰
  REFERRAL = '4'     // 추천인
}

// 주문 상태 타입
export enum OrderStatus {
  PAYMENT_PENDING = 1,   // 결제대기
  PAYMENT_COMPLETED = 2, // 결제완료
  CANCELLED = 99         // 취소
}

// 결제 방법 라벨
export const PaymentTypeLabels = {
  [PaymentType.NEW_SIGNUP]: '신규가입',
  [PaymentType.APP_PAYMENT]: '앱결제',
  [PaymentType.COUPON]: '쿠폰',
  [PaymentType.REFERRAL]: '추천인'
};

// 주문 상태 라벨
export const OrderStatusLabels = {
  [OrderStatus.PAYMENT_PENDING]: '결제대기',
  [OrderStatus.PAYMENT_COMPLETED]: '결제완료',
  [OrderStatus.CANCELLED]: '취소'
}; 

File: backend/app/schemas/order.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum

class PaymentType(str, Enum):
    NEW_SIGNUP = "1"  # 신규가입
    APP_PAYMENT = "2"  # 앱결제
    COUPON = "3"  # 쿠폰
    REFERRAL = "4"  # 추천인

class OrderStatus(int, Enum):
    PAYMENT_PENDING = 1  # 결제대기
    PAYMENT_COMPLETED = 2  # 결제완료
    CANCELLED = 99  # 취소

class OrderBase(BaseModel):
    ot_code: Optional[str] = None
    ot_ori_code: Optional[str] = None
    mt_idx: Optional[int] = None
    mt_id: Optional[str] = None
    ot_title: Optional[str] = None
    ot_pay_type: Optional[str] = None
    ot_sdate: Optional[datetime] = None
    ot_edate: Optional[datetime] = None
    ot_status: Optional[int] = None
    ot_sprice: Optional[float] = None
    ot_use_coupon: Optional[float] = None
    ot_coupon_idx: Optional[int] = None
    ot_price: Optional[float] = None
    ot_price_b: Optional[float] = None
    ot_pay_bank: Optional[str] = None
    ot_rsp_imp_uid: Optional[str] = None
    ct_cancel_amount: Optional[float] = None
    ct_cancel_reson: Optional[str] = None
    ct_refund_holder: Optional[str] = None
    ct_refund_bank: Optional[str] = None
    ct_refund_account: Optional[str] = None
    ot_wdate: Optional[datetime] = None
    ot_pdate: Optional[datetime] = None
    ot_cdate: Optional[datetime] = None
    ot_ccdate: Optional[datetime] = None
    ot_refund_info: Optional[str] = None
    ot_show: Optional[str] = None
    ot_memo: Optional[str] = None
    ot_udate: Optional[datetime] = None
    ot_cash_name: Optional[str] = None
    ot_cash_hp: Optional[str] = None
    ot_cash_chk: Optional[str] = None
    vbank_date: Optional[datetime] = None

class OrderCreate(OrderBase):
    pass

class OrderUpdate(OrderBase):
    pass

class OrderResponse(OrderBase):
    ot_idx: int
    
    class Config:
        from_attributes = True

class OrderListResponse(BaseModel):
    orders: List[OrderResponse]
    total: int
    page: int
    size: int
    total_pages: int

class OrderSummary(BaseModel):
    total_orders: int
    total_amount: float
    completed_orders: int
    pending_orders: int
    cancelled_orders: int
    this_month_amount: float
    average_amount: float

class OrderFilter(BaseModel):
    status: Optional[int] = None
    pay_type: Optional[str] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    page: int = 1
    size: int = 20 